From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lovely_xianxian <xyx_dada@outlook.com>
Date: Fri, 5 Aug 2022 21:34:36 +0800
Subject: [PATCH] EventPriority.NEVER


diff --git a/src/main/java/org/bukkit/event/EventPriority.java b/src/main/java/org/bukkit/event/EventPriority.java
index 376c5375df21d59c520c99eb3bc64e9ddae007a6..da9544b7c3ef155855af74695ec04c72004782dc 100644
--- a/src/main/java/org/bukkit/event/EventPriority.java
+++ b/src/main/java/org/bukkit/event/EventPriority.java
@@ -39,7 +39,11 @@ public enum EventPriority {
      * <p>
      * No modifications to the event should be made under this priority
      */
-    MONITOR(5);
+    MONITOR(5),
+    /**
+     * Event will never be called. Used for modified EventPriorities.
+     */
+    NEVER(-100); // Nebula - EventPriority.NEVER
 
     private final int slot;
 
diff --git a/src/main/java/org/bukkit/event/HandlerList.java b/src/main/java/org/bukkit/event/HandlerList.java
index 2292bd460ce2be113beb4ba6b4eb19350060f01c..7eef7cc6de29e2524d6b245a3ed2e19de4ae6c89 100644
--- a/src/main/java/org/bukkit/event/HandlerList.java
+++ b/src/main/java/org/bukkit/event/HandlerList.java
@@ -122,6 +122,8 @@ public class HandlerList {
      * @param listener listener to register
      */
     public synchronized void register(@NotNull RegisteredListener listener) {
+        if(listener.getPriority() == EventPriority.NEVER) // Nebula - EventPriority.NEVER
+            return;
         if (handlerslots.get(listener.getPriority()).contains(listener))
             throw new IllegalStateException("This listener is already registered to priority " + listener.getPriority().toString());
         handlers = null;
diff --git a/src/main/java/org/bukkit/plugin/SimplePluginManager.java b/src/main/java/org/bukkit/plugin/SimplePluginManager.java
index fc2dae69165776d08274e34a69962cc70445f411..59ed36f5c25c9932d46246ee54478b7a024fe28b 100644
--- a/src/main/java/org/bukkit/plugin/SimplePluginManager.java
+++ b/src/main/java/org/bukkit/plugin/SimplePluginManager.java
@@ -680,6 +680,8 @@ public final class SimplePluginManager implements PluginManager {
 
     @Override
     public void registerEvent(@NotNull Class<? extends Event> event, @NotNull Listener listener, @NotNull EventPriority priority, @NotNull EventExecutor executor, @NotNull Plugin plugin) {
+        if (priority == EventPriority.NEVER) // Nebula - EventPriority.NEVER
+            return;
         registerEvent(event, listener, priority, executor, plugin, false);
     }
 
