From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lovely_xianxian <xyx_dada@outlook.com>
Date: Mon, 27 Jan 2020 09:42:11 +0800
Subject: [PATCH] Localization API


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index e8414592b3afeb1e5db2b817b8fb7c13e073b9aa..6a1d145fca696cb8a560f8739c9e562d4edccd17 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -1963,4 +1963,11 @@ public final class Bukkit {
     public static Server.Spigot spigot() {
         return server.spigot();
     }
+    
+    // Nebula start - Localization API
+    @NotNull
+    public static xianxian.mc.nebula.localization.Localization getLocalization() {
+        return server.getLocalization();
+    }
+    //Nebula end
 }
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index f55ae8275c297c4c86215fba8d7197ffe9715879..432e30190c85ccdeed75f8fea4e9627c2d2ee638 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1724,4 +1724,13 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
     @NotNull
     io.papermc.paper.datapack.DatapackManager getDatapackManager();
     // Paper end
+    
+    // Nebula start - Localization API
+    /**
+     * Get a Localization object, used to fetch localized name of a game object (Items, Entities, Blocks)
+     * @return Localization object
+     */
+    @NotNull
+    xianxian.mc.nebula.localization.Localization getLocalization();
+    // Nebula end
 }
diff --git a/src/main/java/xianxian/mc/nebula/localization/Localization.java b/src/main/java/xianxian/mc/nebula/localization/Localization.java
new file mode 100644
index 0000000000000000000000000000000000000000..a8152033a7163f5ff0950d371ee27c5ea01d8bca
--- /dev/null
+++ b/src/main/java/xianxian/mc/nebula/localization/Localization.java
@@ -0,0 +1,31 @@
+package xianxian.mc.nebula.localization;
+
+import org.bukkit.block.Block;
+import org.bukkit.entity.Entity;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.Nullable;
+
+public interface Localization {
+    
+    /**
+     * Fetch item's localized display name
+     * @param item The item to fetch
+     * @return item's display name, can be null
+     */
+    @Nullable
+    String getLocalizedItemName(@Nullable ItemStack item);
+    /**
+     * Fetch entity's localized display name
+     * @param entity The entity to fetch
+     * @return Entity's display name, can be null
+     */
+    @Nullable
+    String getLocalizedEntityName(@Nullable Entity entity);
+    /**
+     * Fetch block's localized display name
+     * @param block The block to fetch
+     * @return Block's display name, can be null
+     */
+    @Nullable
+    String getLocalizedBlockName(@Nullable Block block);
+}
