From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lovely_xianxian <xyx_dada@outlook.com>
Date: Wed, 11 Mar 2020 19:54:44 +0800
Subject: [PATCH] Rework Daemon Patch


diff --git a/src/main/java/co/aikar/timings/SimpleTimingsReportListener.java b/src/main/java/co/aikar/timings/SimpleTimingsReportListener.java
new file mode 100644
index 0000000000000000000000000000000000000000..5c8bbb6ae64a597a4773ccb152f18a9dcc244617
--- /dev/null
+++ b/src/main/java/co/aikar/timings/SimpleTimingsReportListener.java
@@ -0,0 +1,13 @@
+package co.aikar.timings;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public abstract class SimpleTimingsReportListener extends TimingsReportListener {
+    @Override
+    public void done(@Nullable String url) {}
+    @Override
+    public void sendMessage(@NotNull String message) {}
+    @Override
+    public void addConsoleIfNeeded() {}
+}
diff --git a/src/main/java/co/aikar/timings/TimingsReportListener.java b/src/main/java/co/aikar/timings/TimingsReportListener.java
index ecd149157d4fb80444f34bf5633d74bcdb63dec5..d1a534c6e99a5697f69581e3e378ade49562a89c 100644
--- a/src/main/java/co/aikar/timings/TimingsReportListener.java
+++ b/src/main/java/co/aikar/timings/TimingsReportListener.java
@@ -19,6 +19,11 @@ public class TimingsReportListener implements net.kyori.adventure.audience.Forwa
     private final List<CommandSender> senders;
     private final Runnable onDone;
     private String timingsURL;
+    
+    TimingsReportListener() {
+        senders = Lists.newArrayList();
+        onDone = null;
+    }
 
     public TimingsReportListener(@NotNull CommandSender senders) {
         this(senders, null);
