From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lovely_xianxian <xyx_dada@outlook.com>
Date: Sat, 6 Aug 2022 12:52:50 +0800
Subject: [PATCH] Inconsistent Entity


diff --git a/src/main/java/org/bukkit/entity/Entity.java b/src/main/java/org/bukkit/entity/Entity.java
index 6b842453589cf148ab32c1507cf374056826316e..67d7b5c26cb03e73c254aa6e0b7073f7e42a6056 100644
--- a/src/main/java/org/bukkit/entity/Entity.java
+++ b/src/main/java/org/bukkit/entity/Entity.java
@@ -965,4 +965,22 @@ public interface Entity extends Metadatable, CommandSender, Nameable, Persistent
      */
     @NotNull io.papermc.paper.threadedregions.scheduler.EntityScheduler getScheduler();
     // Paper end - Folia schedulers
+    // Nebula start - Inconsistent Entity
+
+    /**
+     * Indicates whether this entity is an inconsistent entity.
+     * <p>
+     * Inconsistent entities are entities of temporary usages such as {@link ArmorStand} used as chair.
+     * These entities won't be saved to local storage, and may be removed after the chunk containing
+     * it becomes unloaded.
+     *
+     * @return inconsistent or not
+     */
+    boolean isInconsistent();
+
+    /**
+     * Marks an entity as inconsistent. See also: {@link #isInconsistent}
+     */
+    void setInconsistent(boolean inconsistent);
+    // Nebula end
 }
