From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lovely_xianxian <xyx_dada@outlook.com>
Date: Sat, 25 Apr 2020 20:08:43 +0800
Subject: [PATCH] Toggle Sending Chunks API


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 27848ad0eee667e515685a5f1cef3e9bfc7a3f53..1c729a038078f152a4be68526869665ffecaae1b 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -2139,4 +2139,20 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
     @Override
     Spigot spigot();
     // Spigot end
+
+    // Nebula start - Toggle Sending chunks API
+
+    /**
+     * Sets whether send chunks to the player or not
+     * @param sendChunks True if need send chunks to the player
+     */
+    void setSendChunks(boolean sendChunks);
+
+    /**
+     * Gets whether send chunks to the player or not
+     * @return True if need send chunks to the player
+     */
+    boolean needSendChunks();
+    // Nebula end
+
 }
