From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lovely_xianxian <xyx_dada@outlook.com>
Date: Sat, 25 Apr 2020 20:08:43 +0800
Subject: [PATCH] Toggle Sending Chunks API


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 4a4f041c6acfd335871fa67d6d1c7f723229c8a7..83bc78277cb7e8e043f63a4ac633c65a2a6b8b1b 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -2131,4 +2131,20 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
     @Override
     Spigot spigot();
     // Spigot end
+
+    // Nebula start - Toggle Sending chunks API
+
+    /**
+     * Sets whether send chunks to the player or not
+     * @param sendChunks True if need send chunks to the player
+     */
+    void setSendChunks(boolean sendChunks);
+
+    /**
+     * Gets whether send chunks to the player or not
+     * @return True if need send chunks to the player
+     */
+    boolean needSendChunks();
+    // Nebula end
+
 }
