From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lovely_xianxian <xyx_dada@outlook.com>
Date: Fri, 5 Aug 2022 20:09:27 +0800
Subject: [PATCH] Codename Nebula


diff --git a/build.gradle.kts b/build.gradle.kts
index ee49dd4f80ebdfc65cbf860e62d3c80796c8af2c..4b080f5ddd10904387f31a8c949bdcb879bce6fe 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -13,8 +13,8 @@ configurations.named(log4jPlugins.compileClasspathConfigurationName) {
 val alsoShade: Configuration by configurations.creating
 
 dependencies {
-    implementation(project(":paper-api"))
-    implementation(project(":paper-mojangapi"))
+    implementation(project(":nebula-api"))
+    implementation("io.papermc.paper:paper-mojangapi:1.20.1-R0.1-SNAPSHOT")
     // Paper start
     implementation("org.jline:jline-terminal-jansi:3.21.0")
     implementation("net.minecrell:terminalconsoleappender:1.3.0")
@@ -72,7 +72,7 @@ tasks.jar {
         attributes(
             "Main-Class" to "org.bukkit.craftbukkit.Main",
             "Implementation-Title" to "CraftBukkit",
-            "Implementation-Version" to "git-Paper-$implementationVersion",
+            "Implementation-Version" to "git-Nebula-$implementationVersion",
             "Implementation-Vendor" to date, // Paper
             "Specification-Title" to "Bukkit",
             "Specification-Version" to project.version,
@@ -95,6 +95,7 @@ tasks.compileJava {
 publishing {
     publications.create<MavenPublication>("maven") {
         artifact(tasks.shadowJar)
+        artifact(tasks.reobfJar)
     }
 }
 
