From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lovely_xianxian <xyx_dada@outlook.com>
Date: Fri, 27 Dec 2019 21:53:33 +0800
Subject: [PATCH] Configurable ItemFrame breaking in glass


diff --git a/src/main/java/net/minecraft/world/entity/decoration/EntityItemFrame.java b/src/main/java/net/minecraft/world/entity/decoration/EntityItemFrame.java
index 43152a6c70c9433d627a58051101530ddd693307..2057e4d2256ecefd507db4785d0d4d6ecf0f8399 100644
--- a/src/main/java/net/minecraft/world/entity/decoration/EntityItemFrame.java
+++ b/src/main/java/net/minecraft/world/entity/decoration/EntityItemFrame.java
@@ -126,6 +126,9 @@ public class EntityItemFrame extends EntityHanging {
 
     @Override
     public boolean survives() {
+        if (!this.world.nebulaConfig.itemFrameBreakInGlass && this.world.getType(this.blockPosition).getBlock() instanceof net.minecraft.world.level.block.BlockGlass) {
+            return true;
+        }
         if (this.fixed) {
             return true;
         } else if (!this.world.getCubes(this)) {
diff --git a/src/main/java/xianxian/mc/nebula/NebulaWorldConfig.java b/src/main/java/xianxian/mc/nebula/NebulaWorldConfig.java
index bd0103ed8180439e7648a570a4024856ee44ec0b..2e4e743c9a33e516d18491e1dda9cf21ad7495d4 100644
--- a/src/main/java/xianxian/mc/nebula/NebulaWorldConfig.java
+++ b/src/main/java/xianxian/mc/nebula/NebulaWorldConfig.java
@@ -73,4 +73,9 @@ public class NebulaWorldConfig {
     	}
     	log("Phantom spawn rate: " + (phantomSpawnRate == -1 ? "No phantom will spawn" : (phantomSpawnRate + "%")));
     }
+
+    public boolean itemFrameBreakInGlass = true;
+    private void itemFrameBreakInGlass() {
+        itemFrameBreakInGlass = getBoolean("item-frame-break-in-glass", true);
+    }
 }
\ No newline at end of file
