From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lovely_xianxian <xyx_dada@outlook.com>
Date: Fri, 5 Aug 2022 22:15:19 +0800
Subject: [PATCH] Fire EntityMarkDeadEvent


diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index 4c298406ce06d39f8279049270c88ec0830733b1..03201d25f0b8d7fe50e87d784b572cf250605693 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -696,6 +696,7 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource {
 
     public void remove(Entity.RemovalReason reason) {
         this.setRemoved(reason);
+        org.staroceanmc.nebula.event.entity.EntityMarkDeadEvent.callEvent(this.getBukkitEntity());
     }
 
     public void onClientRemoval() {}
