From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lovely_xianxian <xyx_dada@outlook.com>
Date: Fri, 3 Jan 2020 21:45:32 +0800
Subject: [PATCH] Configurable Language


diff --git a/src/main/java/net/minecraft/locale/LocaleLanguage.java b/src/main/java/net/minecraft/locale/LocaleLanguage.java
index 5218214225b50ac4059ab704086a457318e93e00..2c8891fe587574893cc61ad9703a8268a0d6d4ba 100644
--- a/src/main/java/net/minecraft/locale/LocaleLanguage.java
+++ b/src/main/java/net/minecraft/locale/LocaleLanguage.java
@@ -64,7 +64,12 @@ public abstract class LocaleLanguage {
         return new LocaleLanguage() {
             @Override
             public String a(String s) {
-                return (String) map.getOrDefault(s, s);
+                // Nebula start - Configurable Language
+                String s1 = xianxian.mc.nebula.CustomLocaleLanguage.getInstance().translateKey(s);
+                if (s1 == null)
+                    s1 = (String) map.getOrDefault(s, s);
+                return s1;
+                // Nebula end
             }
 
             @Override
diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 35ecd366cdcd5f494099ba49e8f7fe55e0d67e35..6b8bb15668d3f6244e45d1c32330453a4e93d571 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -225,6 +225,10 @@ public class DedicatedServer extends MinecraftServer implements IMinecraftServer
         }
         xianxian.mc.nebula.NebulaConfig.registerCommands();
         // Nebula end
+        // Nebula start - Load custom language on server startup
+        if (xianxian.mc.nebula.NebulaConfig.useCustomLanguageFile)
+            xianxian.mc.nebula.CustomLocaleLanguage.getInstance().load(xianxian.mc.nebula.NebulaConfig.customLanguageFilePath);
+        // Nebula end
         this.setPVP(dedicatedserverproperties.pvp);
         this.setAllowFlight(dedicatedserverproperties.allowFlight);
         this.setResourcePack(dedicatedserverproperties.resourcePack, this.ba());
diff --git a/src/main/java/xianxian/mc/nebula/CustomLocaleLanguage.java b/src/main/java/xianxian/mc/nebula/CustomLocaleLanguage.java
new file mode 100644
index 0000000000000000000000000000000000000000..41fb65762882045dba0e8ac65d64cd2eca9a727e
--- /dev/null
+++ b/src/main/java/xianxian/mc/nebula/CustomLocaleLanguage.java
@@ -0,0 +1,72 @@
+package xianxian.mc.nebula;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.regex.Pattern;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import net.minecraft.util.ChatDeserializer;
+
+/**
+ * See also {@link net.minecraft.locale.LocaleLanguage}
+ */
+public final class CustomLocaleLanguage {
+    private static final CustomLocaleLanguage INSTANCE = new CustomLocaleLanguage();
+    private static final Logger LOGGER = LogManager.getLogger();
+    private static final Pattern PATTERN = Pattern.compile("%(\\d+\\$)?[\\d\\.]*[df]");
+    private static final Gson GSON = new Gson();
+
+    private final Map<String, String> translationMapping = new HashMap<>();
+
+    public void load(String file) {
+        File localeFile = new File(file);
+        if (!localeFile.exists()) {
+            return;
+        }
+
+        long startTime = System.currentTimeMillis();
+        try (FileInputStream fis = new FileInputStream(localeFile)) {
+            JsonElement jsonElement = GSON.fromJson(new InputStreamReader(fis), JsonElement.class);
+            JsonObject jsonobject = ChatDeserializer.m(jsonElement, "strings");
+            Iterator<Entry<String, JsonElement>> iterator = jsonobject.entrySet().iterator();
+
+            while (iterator.hasNext()) {
+                Entry<String, JsonElement> entry = iterator.next();
+                String s = PATTERN.matcher(ChatDeserializer.a(entry.getValue(), entry.getKey())).replaceAll("%$1s");
+
+                this.translationMapping.put(entry.getKey(), s);
+            }
+
+            LOGGER.info("Successfully loaded custom language from " + file + ", cost "
+                    + (System.currentTimeMillis() - startTime) + "ms");
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            LOGGER.error("Couldn't read strings from " + file, e);
+        }
+    }
+
+    public synchronized String translateKey(String key) {
+        if (NebulaConfig.useCustomLanguageFile)
+            return translationMapping.get(key);
+        else
+            return null;
+    }
+
+    public static CustomLocaleLanguage getInstance() {
+        return INSTANCE;
+    }
+}
diff --git a/src/main/java/xianxian/mc/nebula/NebulaConfig.java b/src/main/java/xianxian/mc/nebula/NebulaConfig.java
index 42dc78ca173162a60f002bd5cb797b8945d07a58..1354501cb42827cdd0701fd0a839ba855d9d78a7 100644
--- a/src/main/java/xianxian/mc/nebula/NebulaConfig.java
+++ b/src/main/java/xianxian/mc/nebula/NebulaConfig.java
@@ -174,4 +174,16 @@ public class NebulaConfig {
         config.addDefault(path, def);
         return config.getString(path, config.getString(path));
     }
+
+    //#################### CONFIG AREA ####################
+
+    public static boolean useCustomLanguageFile = false;
+    private static void useCustomLanguageFile() {
+        useCustomLanguageFile = getBoolean("settings.use-custom-language-file", false);
+    }
+
+    public static String customLanguageFilePath = "custom-language.json";
+    private static void customLanguageFilePath() {
+        customLanguageFilePath = getString("settings.custom-language-file-path", "custom-language.json");
+    }
 }
\ No newline at end of file
