From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lovely_xianxian <xyx_dada@outlook.com>
Date: Wed, 8 Jan 2020 22:04:01 +0800
Subject: [PATCH] Show amount of bees in beehives


diff --git a/src/main/java/net/minecraft/world/level/block/BlockBeehive.java b/src/main/java/net/minecraft/world/level/block/BlockBeehive.java
index 828d231a963f1962d88fe170ac86590d86e1df40..b024d00b601698f365506a7d502a5ee6ebbdbca5 100644
--- a/src/main/java/net/minecraft/world/level/block/BlockBeehive.java
+++ b/src/main/java/net/minecraft/world/level/block/BlockBeehive.java
@@ -235,6 +235,16 @@ public class BlockBeehive extends BlockTileEntity {
                 if (flag) {
                     nbttagcompound = new NBTTagCompound();
                     nbttagcompound.set("Bees", tileentitybeehive.m());
+                    // Nebula start - Show amount of bees in beehives
+                    if (xianxian.mc.nebula.NebulaConfig.showBeesAmountInsideBeehive) {
+                        NBTTagCompound display = new NBTTagCompound();
+                        net.minecraft.nbt.NBTTagList lore = new net.minecraft.nbt.NBTTagList();
+                        lore.add(net.minecraft.nbt.NBTTagString.a(net.minecraft.network.chat.IChatBaseComponent.ChatSerializer.a(new net.minecraft.network.chat.ChatComponentText("§7" +
+                                String.format(xianxian.mc.nebula.NebulaConfig.beesInsideLore, tileentitybeehive.getBeeCount())))));
+                        display.set("Lore", lore);
+                        itemstack.a("display", display);
+                    }
+                    // Nebula end
                     itemstack.a("BlockEntityTag", (NBTBase) nbttagcompound);
                 }
 
@@ -265,7 +275,29 @@ public class BlockBeehive extends BlockTileEntity {
             }
         }
 
-        return super.a(iblockdata, loottableinfo_builder);
+        // Nebula start - Show amount of bees in beehives
+        TileEntity tileentity = (TileEntity) loottableinfo_builder.b(LootContextParameters.BLOCK_ENTITY);
+        List<ItemStack> loots = super.a(iblockdata, loottableinfo_builder);
+        if (xianxian.mc.nebula.NebulaConfig.showBeesAmountInsideBeehive)
+            if (tileentity instanceof TileEntityBeehive) {
+                TileEntityBeehive tileentitybeehive = (TileEntityBeehive) tileentity;
+                if (!tileentitybeehive.isEmpty()) {
+                    for (int i = 0, size = loots.size(); i < size; i++) {
+                        // Add lore for each drop
+                        ItemStack loot = loots.get(i);
+                        if (loot.getItem().equals(Items.rr) || loot.getItem().equals(Items.rs)) {
+                            NBTTagCompound display = new NBTTagCompound();
+                            net.minecraft.nbt.NBTTagList lore = new net.minecraft.nbt.NBTTagList();
+                            lore.add(net.minecraft.nbt.NBTTagString.a(net.minecraft.network.chat.IChatBaseComponent.ChatSerializer.a(new net.minecraft.network.chat.ChatComponentText("§7" +
+                                    String.format(xianxian.mc.nebula.NebulaConfig.beesInsideLore, tileentitybeehive.getBeeCount())))));
+                            display.set("Lore", lore);
+                            loot.a("display", display);
+                        }
+                    }
+                }
+            }
+        return loots;
+        // Nebula end
     }
 
     @Override
diff --git a/src/main/java/xianxian/mc/nebula/NebulaConfig.java b/src/main/java/xianxian/mc/nebula/NebulaConfig.java
index 1354501cb42827cdd0701fd0a839ba855d9d78a7..5631e4c88e9ec830af2a05b778dbfb10dc2a7418 100644
--- a/src/main/java/xianxian/mc/nebula/NebulaConfig.java
+++ b/src/main/java/xianxian/mc/nebula/NebulaConfig.java
@@ -186,4 +186,14 @@ public class NebulaConfig {
     private static void customLanguageFilePath() {
         customLanguageFilePath = getString("settings.custom-language-file-path", "custom-language.json");
     }
+
+    public static String beesInsideLore = "%d Bee(s) inside";
+    private static void beesInsideLore() {
+        beesInsideLore = getString("localization.bees-inside-lore", beesInsideLore);
+    }
+
+    public static boolean showBeesAmountInsideBeehive = true;
+    private static void showBeesAmountInsideBeehive() {
+        showBeesAmountInsideBeehive = getBoolean("show-bees-amount-inside-beehive", true);
+    }
 }
\ No newline at end of file
