From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lovely_xianxian <xyx_dada@outlook.com>
Date: Sat, 18 Jan 2020 19:26:09 +0800
Subject: [PATCH] Wolves won't stand up caused by other players


diff --git a/src/main/java/net/minecraft/world/entity/animal/EntityWolf.java b/src/main/java/net/minecraft/world/entity/animal/EntityWolf.java
index b44b1544f401c1a5127bed3239bfd60420d17329..d8358210157dabd4f0d799a2eb87f0845acc6d6d 100644
--- a/src/main/java/net/minecraft/world/entity/animal/EntityWolf.java
+++ b/src/main/java/net/minecraft/world/entity/animal/EntityWolf.java
@@ -326,6 +326,10 @@ public class EntityWolf extends EntityTameableAnimal implements IEntityAngerable
             return flag ? EnumInteractionResult.CONSUME : EnumInteractionResult.PASS;
         } else {
             if (this.isTamed()) {
+                // Nebula start - Wolves won't stand up caused by other players
+                if (!this.j((EntityLiving) entityhuman))
+                    return super.a(entityhuman, enumhand);
+                // Nebula end
                 if (this.k(itemstack) && this.getHealth() < this.getMaxHealth()) {
                     if (!entityhuman.abilities.canInstantlyBuild) {
                         itemstack.subtract(1);
