From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lovely_xianxian <xyx_dada@outlook.com>
Date: Sat, 22 Feb 2020 16:53:40 +0800
Subject: [PATCH] Add option to configure timings url


diff --git a/src/main/java/co/aikar/timings/TimingsExport.java b/src/main/java/co/aikar/timings/TimingsExport.java
index 35810f42d7a0cd50a4cbe90e8d698fe57914c889..0ac8a8d6f132f0d0bb56e53e654c3d2ea4a7a6c5 100644
--- a/src/main/java/co/aikar/timings/TimingsExport.java
+++ b/src/main/java/co/aikar/timings/TimingsExport.java
@@ -307,7 +307,7 @@ public class TimingsExport extends Thread {
         String response = null;
         String timingsURL = null;
         try {
-            HttpURLConnection con = (HttpURLConnection) new URL("http://timings.aikar.co/post").openConnection();
+            HttpURLConnection con = (HttpURLConnection) new URL(xianxian.mc.nebula.NebulaConfig.timingsURL).openConnection();
             con.setDoOutput(true);
             String hostName = "BrokenHost";
             try {
diff --git a/src/main/java/xianxian/mc/nebula/NebulaConfig.java b/src/main/java/xianxian/mc/nebula/NebulaConfig.java
index 5631e4c88e9ec830af2a05b778dbfb10dc2a7418..6639cb5db18a5bd7f6145b63925272ad6a3e7166 100644
--- a/src/main/java/xianxian/mc/nebula/NebulaConfig.java
+++ b/src/main/java/xianxian/mc/nebula/NebulaConfig.java
@@ -196,4 +196,9 @@ public class NebulaConfig {
     private static void showBeesAmountInsideBeehive() {
         showBeesAmountInsideBeehive = getBoolean("show-bees-amount-inside-beehive", true);
     }
+
+    public static String timingsURL = "http://timings.aikar.co/post";
+    private static void timingsURL() {
+        timingsURL = getString("timings.url", timingsURL);
+    }
 }
\ No newline at end of file
