From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lovely_xianxian <xyx_dada@outlook.com>
Date: Wed, 1 Apr 2020 12:18:45 +0800
Subject: [PATCH] Fire PlayerPostTeleportEvent


diff --git a/src/main/java/net/minecraft/server/level/EntityPlayer.java b/src/main/java/net/minecraft/server/level/EntityPlayer.java
index c813077455a463dd558076d1d7474829f76b905a..51c3ed3859fb7f0704fc854b76de66394a23a277 100644
--- a/src/main/java/net/minecraft/server/level/EntityPlayer.java
+++ b/src/main/java/net/minecraft/server/level/EntityPlayer.java
@@ -1306,6 +1306,7 @@ public class EntityPlayer extends EntityHuman implements ICrafting {
                 PlayerChangedWorldEvent changeEvent = new PlayerChangedWorldEvent(this.getBukkitEntity(), worldserver1.getWorld());
                 this.world.getServer().getPluginManager().callEvent(changeEvent);
                 // CraftBukkit end
+                xianxian.mc.nebula.event.player.PlayerPostTeleportEvent.callEvent(tpEvent); // Nebula - Fire PlayerPostTeleportEvent
             }
             // Paper start
             if (this.isBlocking()) {
diff --git a/src/main/java/net/minecraft/server/network/PlayerConnection.java b/src/main/java/net/minecraft/server/network/PlayerConnection.java
index 65b3f59ba741f641ed7d97bd45f37b933140870f..eab97db58a4d26d0a55495bed24d56169f759016 100644
--- a/src/main/java/net/minecraft/server/network/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/network/PlayerConnection.java
@@ -1607,6 +1607,7 @@ public class PlayerConnection implements PacketListenerPlayIn {
         }
 
         this.internalTeleport(d0, d1, d2, f, f1, set);
+        xianxian.mc.nebula.event.player.PlayerPostTeleportEvent.callEvent(event); // Nebula - Fire PlayerPostTeleportEvent
         return event.isCancelled(); // CraftBukkit - Return event status
     }
 
diff --git a/src/main/java/net/minecraft/world/entity/projectile/EntityEnderPearl.java b/src/main/java/net/minecraft/world/entity/projectile/EntityEnderPearl.java
index efdf5bc4f07231690c680429b3f77bd07970eee0..31db1309ff5ecefe370490c2018c368dff756a96 100644
--- a/src/main/java/net/minecraft/world/entity/projectile/EntityEnderPearl.java
+++ b/src/main/java/net/minecraft/world/entity/projectile/EntityEnderPearl.java
@@ -86,6 +86,7 @@ public class EntityEnderPearl extends EntityProjectileThrowable {
                         CraftEventFactory.entityDamage = this;
                         entity.damageEntity(DamageSource.FALL, 5.0F);
                         CraftEventFactory.entityDamage = null;
+                        xianxian.mc.nebula.event.player.PlayerPostTeleportEvent.callEvent(teleEvent); // Nebula - Fire PlayerPostTeleportEvent
                     }
                     // CraftBukkit end
                 }
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index f28f973452ec3eb2d09d0dc1e27e8bb5a04216c1..4c514b4b1f3710ee5057c4acbc60d1b6fdf30e1b 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -929,6 +929,7 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         } else {
             server.getHandle().moveToWorld(entity, toWorld, true, to, !toWorld.paperConfig.disableTeleportationSuffocationCheck); // Paper
         }
+        xianxian.mc.nebula.event.player.PlayerPostTeleportEvent.callEvent(event); // Nebula - Fire PlayerPostTeleportEvent
         return true;
     }
 
