From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lovely_xianxian <xyx_dada@outlook.com>
Date: Tue, 7 Apr 2020 12:10:41 +0800
Subject: [PATCH] Only bedrock and piston head are allowed to be set to air


diff --git a/src/main/java/net/minecraft/world/level/block/piston/BlockPiston.java b/src/main/java/net/minecraft/world/level/block/piston/BlockPiston.java
index 8aa51fb207820a7629d50b80ea821ec6cccf8b54..3b3c7afbe76c35b087108863c432bb030bb31600 100644
--- a/src/main/java/net/minecraft/world/level/block/piston/BlockPiston.java
+++ b/src/main/java/net/minecraft/world/level/block/piston/BlockPiston.java
@@ -260,14 +260,13 @@ public class BlockPiston extends BlockDirectional {
                     }
                 }
             } else {
-                // Paper start - fix headless pistons breaking blocks
-                BlockPosition headPos = blockposition.shift(enumdirection);
-                if (com.destroystokyo.paper.PaperConfig.allowBlockPermanentBreakingExploits || world.getType(headPos) == Blocks.PISTON_HEAD.getBlockData().set(FACING, enumdirection)) { // double check to make sure we're not a headless piston.
-                    world.setAir(headPos, false);
-                } else {
-                    ((WorldServer)world).getChunkProvider().flagDirty(headPos); // ... fix client desync
-                }
-                // Paper end - fix headless pistons breaking blocks
+                // Nebula start - Only bedrock and piston head are allowed to be set to air
+                BlockPosition headPosition = blockposition.shift(enumdirection);
+                IBlockData headData = world.getType(headPosition);
+                Block headBlock = headData.getBlock();
+                if (com.destroystokyo.paper.PaperConfig.allowBlockPermanentBreakingExploits || headBlock == Blocks.BEDROCK || headBlock == Blocks.PISTON_HEAD)
+                // Nebula end
+                world.a(headPosition, false);
             }
 
             world.playSound((EntityHuman) null, blockposition, SoundEffects.BLOCK_PISTON_CONTRACT, SoundCategory.BLOCKS, 0.5F, world.random.nextFloat() * 0.15F + 0.6F);
