From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lovely_xianxian <xyx_dada@outlook.com>
Date: Tue, 7 Apr 2020 12:37:57 +0800
Subject: [PATCH] Fire EntityMarkDeadEvent


diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index 6c4d3aab0a9efcc039071e48d9b9596f1a222694..0d8d9cc4c5a1ac3cb454cb8cb4701971239883b8 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -512,6 +512,7 @@ public abstract class Entity implements INamableTileEntity, ICommandListener, ne
 
     public void die() {
         this.dead = true;
+        xianxian.mc.nebula.event.entity.EntityMarkDeadEvent.callEvent(this.getBukkitEntity());
     }
 
     public void setPose(EntityPose entitypose) {
