From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lovely_xianxian <xyx_dada@outlook.com>
Date: Tue, 7 Apr 2020 12:37:57 +0800
Subject: [PATCH] Fire EntityMarkDeadEvent


diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index feab0ae1930b5271fe0d06a40c180317dcbc9d1d..f2846402aeeb56f166b719cb49329aa979d15900 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -480,6 +480,7 @@ public abstract class Entity implements INamableTileEntity, ICommandListener, ne
 
     public void die() {
         this.dead = true;
+        xianxian.mc.nebula.event.entity.EntityMarkDeadEvent.callEvent(this.getBukkitEntity());
     }
 
     public void setPose(EntityPose entitypose) {
