From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lovely_xianxian <xyx_dada@outlook.com>
Date: Sat, 25 Apr 2020 20:05:29 +0800
Subject: [PATCH] Toggle Sending Chunks API


diff --git a/src/main/java/com/tuinity/tuinity/chunk/PlayerChunkLoader.java b/src/main/java/com/tuinity/tuinity/chunk/PlayerChunkLoader.java
index 0d577aa1c7868ce89c3902535adcb554b1f47551..87a9f98a6b5560f92faeafa6a9ca3bdc7bfc052f 100644
--- a/src/main/java/com/tuinity/tuinity/chunk/PlayerChunkLoader.java
+++ b/src/main/java/com/tuinity/tuinity/chunk/PlayerChunkLoader.java
@@ -490,6 +490,7 @@ public final class PlayerChunkLoader {
                 this.sendingChunkCounts.addTo(data, 1);
             }
 
+            if (data.player.getBukkitEntity().needSendChunks()) // Nebula - Toggle Chunk Sending API
             data.sendChunk(queuedSend.chunkX, queuedSend.chunkZ, () -> {
                 synchronized (this.sendingChunkCounts) {
                     final int count = this.sendingChunkCounts.getInt(data);
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 4c514b4b1f3710ee5057c4acbc60d1b6fdf30e1b..da97d771fb9cc9547d7018005759bcd745027ee0 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -2501,4 +2501,17 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         return spigot;
     }
     // Spigot end
+
+    // Nebula start - Toggle sending chunks API
+    private boolean needSendChunks = true;
+    @Override
+    public void setSendChunks(boolean sendChunks) {
+        this.needSendChunks = true;
+    }
+
+    @Override
+    public boolean needSendChunks() {
+        return needSendChunks;
+    }
+    // Nebula end
 }
