From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lovely_xianxian <xyx_dada@outlook.com>
Date: Sat, 25 Apr 2020 20:13:59 +0800
Subject: [PATCH] Remove debug message from daemon


diff --git a/src/main/java/com/destroystokyo/paper/daemon/PaperDaemon.java b/src/main/java/com/destroystokyo/paper/daemon/PaperDaemon.java
index e0a11e1702ccabaa47ae713b580dc940721628bd..80eedfec3061dec1322e0e104930b27167bdcefa 100644
--- a/src/main/java/com/destroystokyo/paper/daemon/PaperDaemon.java
+++ b/src/main/java/com/destroystokyo/paper/daemon/PaperDaemon.java
@@ -478,7 +478,6 @@ public final class PaperDaemon {
 
     private static void handleMessage(final short messageType, final String message, final int responseChan) {
         final Class<? extends PaperDaemonMessage> clazz = messageTypeMap.get(messageType);
-        log(logger -> logger.warning("handling message"));
         if (clazz == null) {
             log(logger -> logger.warning("Unknown message type: " + messageType));
             return;
