From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lovely_xianxian <xyx_dada@outlook.com>
Date: Sat, 9 May 2020 18:46:50 +0800
Subject: [PATCH] Check illegal fly if player isn't flying


diff --git a/src/main/java/net/minecraft/server/network/PlayerConnection.java b/src/main/java/net/minecraft/server/network/PlayerConnection.java
index 4d6e5f1fa32570e96252c0643401ff32c14c5a01..8951fa1865f6491347c3da4fe84c2b0b4b78e0d4 100644
--- a/src/main/java/net/minecraft/server/network/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/network/PlayerConnection.java
@@ -1502,7 +1502,7 @@ public class PlayerConnection implements PacketListenerPlayIn {
                                 this.player.setLocation(d4, d5, d6, f, f1); // Copied from above
 
                                 // MC-135989, SPIGOT-5564: isRiptiding
-                                this.B = d12 >= -0.03125D && this.player.playerInteractManager.getGameMode() != EnumGamemode.SPECTATOR && !this.minecraftServer.getAllowFlight() && !this.player.abilities.canFly && !this.player.hasEffect(MobEffects.LEVITATION) && !this.player.isGliding() && this.a((Entity) this.player) && !this.player.isRiptiding();
+                                this.B = d12 >= -0.03125D && this.player.playerInteractManager.getGameMode() != EnumGamemode.SPECTATOR && !this.minecraftServer.getAllowFlight() && (!this.player.abilities.canFly || !this.player.abilities.isFlying) && !this.player.hasEffect(MobEffects.LEVITATION) && !this.player.isGliding() && this.a((Entity) this.player) && !this.player.isRiptiding();
                                 // CraftBukkit end
                                 this.player.getWorldServer().getChunkProvider().movePlayer(this.player);
                                 this.player.a(this.player.locY() - d3, packetplayinflying.b());
