From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lovely_xianxian <xyx_dada@outlook.com>
Date: Sun, 10 May 2020 14:50:00 +0800
Subject: [PATCH] Only send entities when needSendChunks is false


diff --git a/src/main/java/net/minecraft/server/level/PlayerChunkMap.java b/src/main/java/net/minecraft/server/level/PlayerChunkMap.java
index d827e18d8c3b9b5869dbb1e233f415ba0efb7c1b..6962554fdec96d50ca14cbc33a72ac9269d60cd5 100644
--- a/src/main/java/net/minecraft/server/level/PlayerChunkMap.java
+++ b/src/main/java/net/minecraft/server/level/PlayerChunkMap.java
@@ -2297,6 +2297,8 @@ Sections go from 0..16. Now whenever a section is not empty, it can potentially
 
     public final void sendChunk(EntityPlayer entityplayer, Packet<?>[] apacket, Chunk chunk) { this.a(entityplayer, apacket, chunk); } // Paper - OBFHELPER
     private void a(EntityPlayer entityplayer, Packet<?>[] apacket, Chunk chunk) {
+        // Nebula start - Only send entities when needSendChunks is false
+        if (entityplayer.getBukkitEntity().needSendChunks()) {
         if (apacket[0] == null) {
             // Paper start - add 8 for light fix workaround
             if (apacket.length != 10) { // in case Plugins call sendChunk, resize
@@ -2366,6 +2368,8 @@ Sections go from 0..16. Now whenever a section is not empty, it can potentially
         // Paper end - Fix MC-162253
 
         entityplayer.a(chunk.getPos(), apacket[0], apacket[1]);
+        }
+        // Nebula end
         PacketDebug.a(this.world, chunk.getPos());
         List<Entity> list = Lists.newArrayList();
         List<Entity> list1 = Lists.newArrayList();
