From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lovely_xianxian <xyx_dada@outlook.com>
Date: Sun, 10 May 2020 14:53:46 +0800
Subject: [PATCH] Do not try to shutdown while shutting down


diff --git a/src/main/java/org/bukkit/craftbukkit/util/ServerShutdownThread.java b/src/main/java/org/bukkit/craftbukkit/util/ServerShutdownThread.java
index c7ed6e0f8a989cec97700df2b15198c9c481c549..d28c83bc4c4bc0aa7dda4f2d349bae1cd06458f6 100644
--- a/src/main/java/org/bukkit/craftbukkit/util/ServerShutdownThread.java
+++ b/src/main/java/org/bukkit/craftbukkit/util/ServerShutdownThread.java
@@ -12,6 +12,7 @@ public class ServerShutdownThread extends Thread {
     @Override
     public void run() {
         try {
+            if (!server.isRunning()) return; // Nebula - Do not try to shutdown while shutting down
             // Paper start - try to shutdown on main
             server.safeShutdown(false, false);
             for (int i = 1000; i > 0 && !server.hasStopped(); i -= 100) {
