From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lovely_xianxian <xyx_dada@outlook.com>
Date: Thu, 21 May 2020 21:54:10 +0800
Subject: [PATCH] Do not allow PaperDaemon.isRestarting to be overwrite


diff --git a/src/main/java/com/destroystokyo/paper/daemon/PaperDaemon.java b/src/main/java/com/destroystokyo/paper/daemon/PaperDaemon.java
index 80eedfec3061dec1322e0e104930b27167bdcefa..715f7f52f5ae8e9007897e75e21ca4e2b9ea9b2f 100644
--- a/src/main/java/com/destroystokyo/paper/daemon/PaperDaemon.java
+++ b/src/main/java/com/destroystokyo/paper/daemon/PaperDaemon.java
@@ -208,7 +208,7 @@ public final class PaperDaemon {
     }
 
     public static void setIsRestarting(final boolean isRestarting) {
-        PaperDaemon.isRestarting = isRestarting;
+        PaperDaemon.isRestarting = PaperDaemon.isRestarting || isRestarting;
     }
 
     /**
diff --git a/src/main/java/org/spigotmc/RestartCommand.java b/src/main/java/org/spigotmc/RestartCommand.java
index de9f31052ac55ed4a3ddb94895161b20482c2c8c..7cac4ce3f1ea2cb2282b651158e4ac80ff381fb5 100644
--- a/src/main/java/org/spigotmc/RestartCommand.java
+++ b/src/main/java/org/spigotmc/RestartCommand.java
@@ -76,6 +76,7 @@ public class RestartCommand extends Command
     // Paper start - sync copied from above with minor changes, async added
     public static void shutdownServer(boolean isRestarting) // Paper - make public
     {
+        com.destroystokyo.paper.daemon.PaperDaemon.setIsRestarting(true); // Paper daemon
         if ( MinecraftServer.getServer().isMainThread() )
         {
             // Kick all players
