From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lovely_xianxian <xyx_dada@outlook.com>
Date: Sat, 19 Sep 2020 12:55:14 +0800
Subject: [PATCH] Fix StackOverFlow in EntityWolf.b


diff --git a/src/main/java/net/minecraft/world/entity/animal/EntityWolf.java b/src/main/java/net/minecraft/world/entity/animal/EntityWolf.java
index d8358210157dabd4f0d799a2eb87f0845acc6d6d..aae63d162673877425d96d98c61aee62773a877a 100644
--- a/src/main/java/net/minecraft/world/entity/animal/EntityWolf.java
+++ b/src/main/java/net/minecraft/world/entity/animal/EntityWolf.java
@@ -328,7 +328,7 @@ public class EntityWolf extends EntityTameableAnimal implements IEntityAngerable
             if (this.isTamed()) {
                 // Nebula start - Wolves won't stand up caused by other players
                 if (!this.j((EntityLiving) entityhuman))
-                    return super.a(entityhuman, enumhand);
+                    return EnumInteractionResult.PASS;
                 // Nebula end
                 if (this.k(itemstack) && this.getHealth() < this.getMaxHealth()) {
                     if (!entityhuman.abilities.canInstantlyBuild) {
